### DISCUSSION TITLE
Some important questions.

### DISCUSSION THREAD

**Discussion by wireboy5** [Slow-Scallion4183/Adventure-Colussus#34] [General]
So, I am starting to work on loading "Adventures" from a file and I need some input (no pun intended).

Here is my concept for storing adventures on disk.

In the adventure, there are rooms. Each room has a description, and contents. The contents can be anything from "encounters" (fights and stuff) to NPCs willing to trade. The game itself describes the room, and asks what you want to do. Talking to an NPC will start the NPC specific dialogue, going through a door will take you to the next room, etc.

Does this sound good for a basic system? It can always be expanded upon, and the upside of this system is we can generate maps of the adventure on the fly. (So your character can have a "look at map" option)

I can start a basic implementation, but I just want some feedback on what you think @Slow-Scallion4183 
Happy Coding :D

**Comment by Slow-Scallion4183** [comment]
Sounds very cool! Its very good for the basic implementation. 

I guessing we may move around the room with North, South, East and West.  (Just a guess). 

Descriptions can be generated from the attributes of a room for example (just an idea). Maybe listing the NPC's in the room, but not the enemys. 

Example of gameplay:

Description: (Description)

NPC's: (Current NPC's in the room)

> Would you like to move North, east, south or west? 
> east

> You move to a wall on your right
> Would you like to move North, east, south or west? 
> North

> You encounter a/an {enemy_name}
> Would you like to move North, east, south or west? 
> South

> You encounter an NPC named {NPC_Name}

etc

just an idea

**Reply by wireboy5** [reply]
👍

**Reply by ghost** [reply]
dudes. literally! wth is this. he said this game is 2d but how can we place chests and room entries when the game itself is 2d?

**Reply by wireboy5** [reply]
The chests will just sorta be NPCs that you can trade with or take from for free. The game will simply tell you when you found a chest, and ask if you want to open this. Everything, including room entries, is a concept that is used in the code, in the background, but the beauty of this game is that it will be able to take you through a small but complete world that can be described as 3D, rendered as 2D, and stored as both. My reasoning is that if @Slow-Scallion4183  ever wants to make a 3d version, adventures from this version should be importable.

**Reply by Slow-Scallion4183** [reply]
thanks for clearing that up for SkepticalPotato2k mate. Very informative.

Here is an example **SkeptikalPotato2k**  how it would work:

```
> Would you like to move North, east, south or west?
> east

> You find a chest! Would you like to open it and have a look what things it holds?
> Yes

> The chest contained a diamond bow! You ranged damage is now {ranged_damage}!
```

**Reply by ghost** [reply]
but where will be the npc placed when theres no space lol.

**Reply by Slow-Scallion4183** [reply]
The rooms will be bigger than just one 'tile'...
